src/nrniv/nonlinz.cpp

```diff
@@ -245,46 +228,21 @@ NonLinImpRep::NonLinImpRep() {
     if (neq_ == 0) {
         return;
     }
-    m_ = cmplx_spCreate(neq_, 1, &err);
-    assert(err == spOKAY);
+    m_.resize(neq_, neq_);
     pv_.resize(neq_);
     pvdot_.resize(neq_);
-    v_index_ = new int[n_v_];
-    rv_ = new double[neq_ + 1];
-    rv_ += 1;
-    jv_ = new double[neq_ + 1];
-    jv_ += 1;
-    diag_ = new double*[neq_];
-    deltavec_ = new double[neq_];
-
-    for (i = 0; i < n_v_; ++i) {
+    v_.resize(neq_);
+    deltavec_.resize(neq_);
+
+    for (int i = 0; i < n_v_; ++i) {
         // utilize nd->eqn_index in case of use_sparse13 later
         Node* nd = _nt->_v_node[i];
         pv_[i] = nd->v_handle();
         pvdot_[i] = nd->rhs_handle();
-        v_index_[i] = i + 1;
-    }
-    for (i = 0; i < n_v_; ++i) {
-        diag_[i] = cmplx_spGetElement(m_, v_index_[i], v_index_[i]);
```

Comment:
for my understanding, why are we not setting the diagonal elements of `m_` instead?

Reply:
Here this is only a place where we get the pointer to the value, later they are set.

Reply:
understood.